source/Messages.cpp

```diff
@@ -33,6 +39,9 @@ void Messages::Add(const string &message, Importance importance)
 {
 	lock_guard<mutex> lock(incomingMutex);
 	incoming.emplace_back(message, importance);
+	logged.emplace(logged.begin(), message, importance);
+	if(logged.size() > MAX_LOG)
+		logged.pop_back();
```

Comment:
Check that the new message isn't the same as the message at the front of the deque. Also, it looks like deque has an emplace_front function that we could use instead of just normal emplace with the begin iterator.
```suggestion
	if(message != logged.front())
	{
		logged.emplace_front(message, importance);
		if(logged.size() > MAX_LOG)
			logged.pop_back();
	}
```